CMakeLists.txt

```diff
@@ -1,154 +1,343 @@
-#===-- CMakeLists.txt ------------------------------------------------------===#
-#
-# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-# See https://llvm.org/LICENSE.txt for license information.
-# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-#
-#===------------------------------------------------------------------------===#
-
 cmake_minimum_required(VERSION 3.9.0)
 
-option(LINK_WITH_FIR "Link driver with FIR and LLVM" ON)
+# RPATH settings on macOS do not affect INSTALL_NAME.
+if (POLICY CMP0068)
+  cmake_policy(SET CMP0068 NEW)
+  set(CMAKE_BUILD_WITH_INSTALL_NAME_DIR ON)
+endif()
 
-# Pass -DGCC=... to cmake to use a specific gcc installation.
-if( GCC )
-  set(CMAKE_CXX_COMPILER "${GCC}/bin/g++")
-  set(CMAKE_CC_COMPILER "${GCC}/bin/gcc")
-  set(CMAKE_BUILD_RPATH "${GCC}/lib64")
-  set(CMAKE_INSTALL_RPATH "${GCC}/lib64")
+# Include file check macros honor CMAKE_REQUIRED_LIBRARIES.
+if(POLICY CMP0075)
+  cmake_policy(SET CMP0075 NEW)
 endif()
-if(BUILD_WITH_CLANG)
-  file(TO_CMAKE_PATH "${BUILD_WITH_CLANG}" CLANG_PATH)
-  set(CMAKE_CXX_COMPILER "${CLANG_PATH}/bin/clang++")
-  set(CMAKE_CC_COMPILER "${CLANG_PATH}/bin/clang")
-  if(GCC)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --gcc-toolchain=${GCC}")
+
+# option() honors normal variables.
+if (POLICY CMP0077)
+  cmake_policy(SET CMP0077 NEW)
+endif()
+
+# Flang requires C++17.
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+set(FLANG_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+
+if (CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR AND NOT MSVC_IDE)
+  message(FATAL_ERROR "In-source builds are not allowed. \
+          Please create a directory and run cmake from there,\
+          passing the path to this source directory as the last argument.\
+          This process created the file `CMakeCache.txt' and the directory\
+          `CMakeFiles'. Please delete them.")
+endif()
+
+# Add Flang-centric modules to cmake path.
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
+include(AddFlang)
+
+
+# Check for a standalone build and configure as appropriate from
+# there.
+if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+message("Building Flang as a standalone project.")
+project(Flang)
+
+  set(FLANG_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
+  if (NOT MSVC_IDE)
+    set(LLVM_ENABLE_ASSERTIONS ${ENABLE_ASSERTIONS}
+      CACHE BOOL "Enable assertions")
+    # Assertions follow llvm's configuration.
+    mark_as_advanced(LLVM_ENABLE_ASSERTIONS)
+  endif()
+
+  # We need a pre-built/installed version of LLVM.
+  find_package(LLVM REQUIRED HINTS "${LLVM_CMAKE_PATH}")
+  list(APPEND CMAKE_MODULE_PATH ${LLVM_DIR})
+
+  # If LLVM links to zlib we need the imported targets so we can too.
+  if(LLVM_ENABLE_ZLIB)
+    find_package(ZLIB REQUIRED)
+  endif()
+
+  set(LLVM_DEFAULT_EXTERNAL_LIT ${LLVM_DIR}/../../../bin/llvm-lit)
```

Comment:
What is this for?

Reply:
No longer necessary and left in by mistake.